src/lookup.py

```diff
@@ -687,162 +700,362 @@ def matches(self, entitytype=None, variant=None, subtype=None):
-                    f'Entity "{name}" from "{modName}" has a type outside the 0 - 999 range! ({entityType}) It will not load properly from rooms!'
-                )
+    def loadEntityNode(
+        self,
+        node: ET.Element,
+        resourcePath="",
+        modName="Basement Renovator",
+        entities2Root=None,
+    ):
+        entityType = node.get("ID")
+        variant = node.get("Variant")
+        subtype = node.get("Subtype")
+        name = node.get("Name")
+
+        if (name is not None) or (entityType is not None):
+            isRef = True
+            for key in node.attrib.keys():
+                if key not in ("ID", "Variant", "Subtype", "Name"):
+                    isRef = False
+                    break
 
-            if variant >= 4096 or variant < 0:
-                printf(
-                    f'Entity "{name}" from "{modName}" has a variant outside the 0 - 4095 range! ({variant})'
-                )
+            if isRef:
+                if entityType:
+                    entityType = int(entityType)
+                if variant:
+                    variant = int(variant)
+                if subtype:
+                    subtype = int(subtype)
 
-            if (subtype >= 256 or subtype < 0) and (
-                entityType != EntityType["PICKUP"]
-                or variant not in EntityLookup.PICKUPS_WITH_SPECIAL_SUBTYPES
-            ):
-                printf(
-                    f'Entity "{name}" from "{modName}" has a subtype outside the 0 - 255 range! ({subtype})'
+                entityConfig = self.lookupOne(
+                    entitytype=entityType, variant=variant, subtype=subtype, name=name
                 )
+                if entityConfig:
+                    return entityConfig
+                else:
+                    printf(
+                        f"Entity {node.attrib} looks like a ref, but was not previously defined!"
+                    )
 
-            entityXML = None
+        entityType = int(entityType or -1)
+        variant = int(variant or 0)
+        subtype = int(subtype or 0)
 
-            if entity.get("Metadata") != "1" and entity.get("IsGrid") != "1":
-                adjustedId = "1000" if entityType == "999" else entityType
-                query = f"entity[@id='{adjustedId}'][@variant='{variant}']"
+        if (entityType >= 1000 or entityType < 0) and node.get("IsGrid") != "1":
+            printf(
+                f'Entity "{name}" from "{modName}" has a type outside the 0 - 999 range! ({entityType}) It will not load properly from rooms!'
+            )
 
-                validMissingSubtype = False
+        if variant >= 4096 or variant < 0:
+            printf(
+                f'Entity "{name}" from "{modName}" has a variant outside the 0 - 4095 range! ({variant})'
+            )
 
-                if entities2Root:
-                    entityXML = entities2Root.find(query + f"[@subtype='{subtype}']")
+        if (subtype >= 256 or subtype < 0) and (
```

Comment:
shouldn't these conditions be replaced by the EntityInfo function you made in the other PR?